src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/SelectForUpdateTest.java

```diff
@@ -0,0 +1,222 @@
+
+    /**
+     * Value to store in the {@link #PK_COL} column.
+     */
+    private static final String PK_COL_VAL = "PK";
+
+    /**
+     * The primary {@link DatabaseEngine} to use in the test.
+     */
+    protected DatabaseEngine engine;
+
+    /**
+     * Useful properties for the {@link DatabaseEngine}.
+     */
+    protected Properties properties;
+
+    /**
+     * Loads the DB configurations to be used in the tests.
+     *
+     * @return The list of DB configurations.
+     * @throws Exception In case configurations can't be loaded.
+     */
+    @Parameterized.Parameters
+    public static Collection<DatabaseConfiguration> data() throws Exception {
+        return DatabaseTestUtil.loadConfigurations();
+    }
+
+    /**
+     * A specific DB config loaded that will be used in the test.
+     */
+    @Parameterized.Parameter
+    public DatabaseConfiguration config;
+
+    @Before
+    public void init() throws Exception {
+        this.properties = new Properties() {
+
+            {
+                setProperty(JDBC, config.jdbc);
+                setProperty(USERNAME, config.username);
+                setProperty(PASSWORD, config.password);
+                setProperty(ENGINE, config.engine);
+                setProperty(SCHEMA_POLICY, "create");
+            }
+        };
+
+        this.engine = DatabaseFactory.getConnection(this.properties);
+
+        // Create table
+        DbEntity testEntity = new DbEntity.Builder()
+                .name(TEST_TABLE)
+                .addColumn(PK_COL, DbColumnType.STRING)
+                .addColumn(COOL_COL, DbColumnType.STRING)
+                .pkFields(PK_COL)
+                .build();
+        this.engine.addEntity(testEntity);
+
+        // Populate with first item
+        this.engine.persist(
+                TEST_TABLE,
+                entry().set(PK_COL, PK_COL_VAL)
+                        .set(COOL_COL, INIT_COOL_COL_VAL)
+                        .build(),
+                false
+        );
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        this.engine.dropEntity(TEST_TABLE);
+        this.engine.close();
+    }
+
+    /**
+     * Tests that when a client performs a select for update on a table previously selected for update, it will block
+     * waiting for the first client to finish the transaction.
+     *
+     * @throws Exception In case anything goes wrong.
+     */
+    @Test(timeout = 20000)
+    public void testThatSecondClientQueryReadsValueOnlyAfterFirstClientCommit() throws Exception {
+        this.engine.beginTransaction();
+        queryForUpdate(this.engine);
+
+        final CountDownLatch secondClientSelectedForUpdate = new CountDownLatch(1);
+        final Future<Boolean> secondClient = CompletableFuture.supplyAsync(() -> {
+            try {
+                final DatabaseEngine engine2 = DatabaseFactory.getConnection(properties);
+                while (true) {
```

Comment:
Put a boolean that you can control here so that in the end of the test this is interrupted.
Otherwise this may be running forever in the jvm.